bark/bark.py

```diff
@@ -218,6 +280,40 @@ def write_metadata(filename, **params):
         yaml_file.write(yaml.safe_dump(params, default_flow_style=False))
 
 
+def _enforce_units(params):
+    """
+    Enforces strict adherence to the BARK spec for the 'units' attribute.
+
+    Modifies the given dict in place.
+    """
+    if 'units' in params:
+        if params['units'] == 'seconds':
+            params['units'] = 's'
+    return
+
+def _enforce_datatype(params):
+    """
+    Checks params['datatype'] against canonical list.
+
+    Throws KeyError if params['datatype'] is not on the list.
+
+    If there is no 'datatype' attribute, infers time series / point process
+    identity from 'units' and assigns either 0/UNDEFINED or 1000/EVENT,
+    respectively.
+
+    Modifies the given dict in place.
+    """
+    if 'datatype' in params:
+        if DataTypes._fromcode(params['datatype']) is None:
+            raise KeyError('bad datatype code: {}'.format(params['datatype']))
+    elif 'units' in params and params['units'] in UNITS.TIME_UNITS:
+        params['datatype_name'] = 'EVENT'
```

Comment:
I'm not sure about adding a new undocumented attribute. It could cause confusion when the datatype is changed but datatype_name says something else.